### ISSUE TITLE
QueryOptimizer does not work when returning Node interface

### ISSUE THREAD

**Issue by yergom** [blb-ventures/strawberry-django-plus#28]
Hi! First of all, congratulations with the library, it's a great addition to strawberry's ecosystem.

I'm trying to implement relay's node query:
```graphql
type Query{
   node(id: ID!): Node
}
```
which I've implemented with:
```python
@gql.type
class Query:
    @gql.field
    def node(info: Info, id: gql.relay.GlobalID) -> gql.Node:
        return id.resolve_node(info, required=True)
```
When executing a query using fragments:

```graphql
{
node(id="id"){
  id
  ... on MyDjangoType{
    fieldA
    fieldB
   }
  } 
}
```
The query optimizer does not seem capable of optimizing it.

**Comment by bellini666** [comment]
Hey @yergom !

I think that the issue is because you defined it using `gql.field` instead of `gql.django.field`. Could you try that?

Also, it should work with just `node: gql.Node = gql.django.node()` (please tell me if it doesn't)

But if you have non django types implementing the `Node` interface, then the resolver would need to workaround that. It should not be complicated 3 to 5 lines of code should do that and I can tell you want exactly needs to be done, but lets see if you really need that first =P

**Comment by yergom** [comment]
I've tried both options, and none of them optimize the query. The reproduction steps are fairly simple. Given the following model:

```python
class Container(models.Model):
    name = models.TextField()

class Item(models.Model):
    name = models.TextField()
    container = models.ForeignKey(Container, related_name="items", on_delete=models.CASCADE)

@gql.django.type(Item)
class ItemType(gql.Node):
    description: gql.auto
    boat: "ContainerType"

@gql.django.type(Container)
class ContainerType(gql.Node):
    name: gql.auto
    areas: List[ItemType]

@gql.type
class Query:
    node: gql.Node = gql.django.node()
    
    @gql.django.connection
    def all_containers(self) -> Iterable[ContainerType]:
        return Container.objects.all()
```

And instantiating 1 container with 10 items, the following query:

```graphql
query all {
  allContainers{
    edges{
      node{
        id
        name
		items{
          description
        }
      }
    }
  }
}
```

requires 3 sql queries (1 for counting, 1 for getting the container id/name, and 1 for the prefetch of Items), whereas the following query:

```graphql
query node{
  node(id:"UD"){
    ... on Container{
      name
      items{
        description
      }
    }
  }
}
```

is creating 13 sql queries. So no `prefetch` and no `only` optimization.

**Comment by bellini666** [comment]
Hey @yergom ,

Just to ask you if you had any luck working around your issue? I unfortunately didn't have much time those days to check this. Will try to do soon though.

**Comment by yergom** [comment]
Hey @bellini666, unfortunately I haven't had time either. For now, my (poor's man) solution is to create independent query fields per type, which is obviously not ideal.

**Comment by bellini666** [comment]
@yergom this should be fixed in the release 1.11 that I'm doing atm. Please reopen if it is not.
